tests/DSL/SearchBuilderTest.php

```diff
@@ -416,6 +416,24 @@ public function it_set_a_filter_query()
     /**
      * @test
      */
+    public function it_overwrites_the_connections_default_index()
+    {
+        $builder = $this->getBuilder();
+        $connection = $builder->getConnection();
+
+        $connection->shouldReceive('setIndex')
+                   ->with('test')
+                   ->set('index', 'test')
```

Comment:
Since connection is a mock we'll have to use mockery's set method to set the value of the index property. This way we can make the assertion, but it isn't really testing the connection class, is it?